### PR TITLE
Feature/comm 2238  create new api

### PR DESCRIPTION
**Pull request by ivanzusko**

### CODE REVIEW

**Comment by Vxplus** [comment]
Can this be splitted in multiple prs?

**Comment by Vxplus** [comment]
also pleas use no edits instead of push amends

File: src/modules/forms/plugins/RequestTransformer/RequestTransformerPlugin.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    return (
+        <div className={b('section')()}>
+            <Row fullwidth>
+                <Row col>
+                    <Label>Plugin Name</Label>
+                    <Input input={{value: 'Request Transformer'}} disabled />
+                </Row>
+                <ControlBar name={`${name}.enabled`} removePlugin={() => handlePluginExclude(pluginName)} />
+            </Row>
+            <Row className={b('row')()} fullwidth>
```
Comment (Vxplus) [inline src/modules/forms/plugins/RequestTransformer/RequestTransformerPlugin.js]: I see this repeats a lot any why to DRY it?

File: src/modules/pages/NewApiPage/NewApiForm.js [hunk 2]
```diff
@@ -19,62 +31,272 @@ const ApiForm = (props) => {
...
+                                    <Label htmlFor="is-not-active">No</Label>
+                                </Row>
+                            </Row>
+                        </Row>
+                    </Row>
+                </div>
+                <div className={b('section')}>
+                    <div className={b('section-title')}>2. Proxy</div>
+                    <Row className={b('row')()} fullwidth>
+                        <Row col>
```
Comment (Vxplus) [inline src/modules/pages/NewApiPage/NewApiForm.js]: Same here this rows could also be taken into a component

File: src/modules/pages/NewApiPage/newApiContainer.js [hunk 3]
```diff
@@ -4,16 +4,19 @@ import {
...
 
 const mapStateToProps = state => ({
     api: state.apiReducer.api,
+    selectedPlugins: state.apiReducer.selectedPlugins,
 });
 
 export default connect(
     mapStateToProps,
-    { fetchEndpointSchema, resetEndpoint, saveEndpoint, willClone },
+    { fetchEndpointSchema, resetEndpoint, saveEndpoint, excludePlugin, selectPlugin, willClone },
```
Comment (Vxplus) [inline src/modules/pages/NewApiPage/newApiContainer.js]: maybe drop them to one lines for readability?

File: src/styles/input.scss [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    &::placeholder {
+        color: $hf-color-gray-medium;
+    }
+
+    &:disabled {
+        background-color: $hf-color-gray-pale;
+    }
+}
+
+.Select {
```
Comment (Vxplus) [inline src/styles/input.scss]: Why uppercase?

Comment (ivanzusko) [reply src/styles/input.scss]: It is the `react-select` library's class name @Vxplus

File: src/modules/pages/NewApiPage/NewApiForm.js (hunk 2 above)
Comment (ivanzusko) [reply src/modules/pages/NewApiPage/NewApiForm.js]: in separate **PR** I will remove all these layout components in favor of `classNames`.

File: src/modules/pages/NewApiPage/newApiContainer.js (hunk 3 above)
Comment (ivanzusko) [reply src/modules/pages/NewApiPage/newApiContainer.js]: `true`

File: src/modules/forms/plugins/RequestTransformer/RequestTransformerPlugin.js (hunk 1 above)
Comment (ivanzusko) [reply src/modules/forms/plugins/RequestTransformer/RequestTransformerPlugin.js]: @Vxplus  the same as with the next comment... will be completely removed

**Comment by ivanzusko** [comment]
@Vxplus deal 😄  you'll be the first whom I'll **ping**

**Pull request merged**